device.go

```diff
@@ -28,3 +32,20 @@ func IsCUDAAvailable() bool {
 func IsCUDNNAvailable() bool {
 	return C.IsCUDNNAvailable() == C.bool(true)
 }
+
+// SetNumThreads sets the number of threads created by GoTorch
+func SetNumThreads(n int32) {
+	C.SetNumThreads(C.int32_t(n))
+}
+
+func init() {
+	// Avoid creating too many threads: the original default setting of OMP_NUM_THREADS
+	// (defaults to the core number) may degrade performance on GPUs.
```

Comment:
Why OMP_NUM_THREADS = core number "may degrade" performance on GPUs?

Why should we change it to #core/2 ?